### PR TITLE
[enh] add test.yamllint - lint yaml files

### PR DESCRIPTION
**Pull request by return42**
Usage::

    make test.yamllint

    ./manage test.yamllint

test.yamllint is also added to the `test` makefile target.

### CODE REVIEW

**Comment by return42** [comment]
@dalf I rebased on top of current master branch / solvde conflicts in `searx/settings.yml`  ..

If you do not have any doubt about linting YAML and you think this PR is OK ... can we merge? ..  continue to solve conflicts in `searx/settings.yml` is a little bit error prone task 😓

File: manage [hunk 1]
```diff
@@ -29,6 +29,11 @@ pylint.FILES() {
     grep -l -r --include \*.py '^#[[:blank:]]*lint:[[:blank:]]*pylint' searx searx_extra tests
 }
 
+YAMLLINT_FILES=()
+while IFS= read -r line; do
+   YAMLLINT_FILES+=("$line")
```
Comment (dalf) [inline manage:34]: It should ignore files defined by `.gitignore`.
More into details: `make themes` will install different NPM packages, some of them includes .yml files which don't follow the yamllint format.

Perhaps `git ls-files '*.yml'` ?

**Comment by dalf** [comment]
What do you think of something like [yamlfmt](https://pypi.org/project/yamlfmt/) ?

**Comment by return42** [comment]
> What do you think of something like yamlfmt ?

- Requires [ruamel-yaml](ruamel.yaml ) in which I have a doubt.
- To have some quality, refactoring of the yaml files should made by handcraft than automated.

File: manage (hunk 1 above)
Comment (return42) [reply manage:34]: fixed to: `git ls-files './tests/*.yml' './searx/*.yml'` / see [9ce7dc6f](https://github.com/searxng/searxng/pull/105/commits/9ce7dc6f2d4b5dd14b0f0ff04ff9f7a333595ef9#diff-181229424893bb65d94a74c2132b8b9e5adfe851464fdb5cb9f49e8a8204be7bR35)

**Comment by return42** [comment]
FYI: I rebased this branch / what has changed? ..

- the first commit (393c06f5) now includes a `.yamllint.yml`
- the ducument start/end markup has been removed
- the `# yamllint disable-line ...` are no longer needed and has been removed

**update**
- commit 6b963309 has a _flycheck_ setup which can be used in emacs

**Pull request merged**